server/src/features/completion.rs

```diff
@@ -565,7 +575,48 @@ fn complete_string_literal(session: &mut SessionInfo, file: &Rc<RefCell<Symbol>>
                     }
                 }
             },
-            ExpectedType::DOMAIN => {
+            ExpectedType::DOMAIN(parent) => {},
+            ExpectedType::DOMAIN_OPERATOR => {
+                for operator in vec!["!", "&", "|"].iter() {
+                    items.push(CompletionItem {
+                        label: operator.to_string(),
+                        insert_text: None,
+                        kind: Some(lsp_types::CompletionItemKind::CLASS),
+                        label_details: None,
+                        sort_text: None,
+                        ..Default::default()
+                    });
+                }
+            },
+            ExpectedType::DOMAIN_LIST(parent) => {},
+            ExpectedType::DOMAIN_COMPARATOR => {
+                for (operator, sort_text) in vec![("=", "a"), ("!=", "b"), (">", "c"), (">=", "d"), ("<", "e"), ("<=", "f"), ("=?", "g"),  ("like", "h"), ("=like", "i"), ("not like", "j"), ("ilike", "k"),
+                    ("=ilike", "l"),  ("not ilike", "m"),  ("in", "n"),  ("not in", "o"), ("child_of", "p"), ("parent_of", "q"), ("any", "r"), ("not any", "s")].iter() {
+                    items.push(CompletionItem {
+                        label: operator.to_string(),
+                        insert_text: None,
+                        kind: Some(lsp_types::CompletionItemKind::CLASS),
+                        label_details: None,
+                        sort_text: Some(sort_text.to_string()),
+                        ..Default::default()
+                    });
+                }
+            },
+            ExpectedType::DOMAIN_FIELD(parent) => {
+                let mut all_symbols: HashMap<String, Vec<(Rc<RefCell<Symbol>>, Option<String>)>> = HashMap::new();
+                let from_module = file.borrow().find_module().clone();
+                Symbol::all_members(&parent, session, &mut all_symbols, true, from_module, &mut None, false);
+                for (_symbol_name, symbols) in all_symbols {
+                    //we could use symbol_name to remove duplicated names, but it would hide functions vs variables
```

Comment:
But we cannot have functions and variables with the same name, can we?

Reply:
We could. Some would be shadowed, but as we are searching here in classes, inheritance, co-models,... we could find different symbols type with the same name